packages/middleware-endpoint/src/service-customizations/s3.ts

```diff
@@ -61,9 +61,10 @@ export const isDnsCompatibleBucketName = (bucketName: string): boolean =>
 export const isArnBucketName = (bucketName: string): boolean => {
   const [arn, partition, service, region, account, typeOrId] = bucketName.split(":");
   const isArn = arn === "arn" && bucketName.split(":").length >= 6;
-  const isValidArn = [arn, partition, service, account, typeOrId].filter(Boolean).length === 5;
+  const isValidArn =
+    [arn, partition, service, typeOrId].filter(Boolean).length === 4 && [region, account].filter(Boolean).length == 0;
```

Comment:
We cannot require that region and account are definitively not set. 

An access point MRAP ARN works with and without the account id present, for example with `listObjects({ Bucket: "arn:aws:s3::12345:accesspoint/xyz.mrap" })`